unistd/file.c

```diff
@@ -120,9 +120,9 @@ int open(const char *filename, int oflag, ...)
 	mode = va_arg(ap, mode_t);
 	va_end(ap);
 
-	if (oflag & (O_WRONLY | O_RDWR)) {
+	if (oflag & (O_WRONLY | O_RDWR | O_RDONLY)) {
 		if ((err = stat(filename, &st)) < 0) {
-			if (errno != ENOENT)
+			if (errno != ENOENT || (oflag & O_RDONLY))
```

Comment:
Why this change? I don't see a correlation with PR message:
"
    Add handling of path = "" in open() for accordance with:

    open() shall fall if:
    [ENOENT] O_CREAT is not set and a component of path does not name an existing file, or O_CREAT is set and a component of the path prefix of path does not name an existing file, or path points to an empty string.
"

Reply:
Yes, this was valid without/before `resolve_path()` changes, sorry for that. 

Change in _unistd/file.c_ dropped.